httpserver-rs/src/settings.rs

```diff
@@ -71,6 +71,9 @@ pub struct ServiceSettings {
     /// cache control options
     pub cache_control: Option<String>,
 
+    /// Flag for read only mode
+    pub readonly_mode: Option<bool>,
+
```

Comment:
Does this have to `#[serde(default)]` to maintain compatibility or is it None by default?

Reply:
I think it would be better to put `#[serde(default)]` in there

Reply:
It's `Some(false)` [by default](https://github.com/aish-where-ya/capability-providers/blob/886a2bc1431a180c4f786e5f5255bd3cec5c55cb/httpserver-rs/src/settings.rs#L102). We don't deserialize this struct unless one of `config_file`, `config_b64`, or `config_json` are provided

We could change the type to be `bool` and simply default to false for clarity, but that's a supernit